Packages/Blockchain/Sources/Ethereum/EthereumFeeService.swift

```diff
@@ -114,10 +115,93 @@ extension EthereumService: ChainFeeCalculateable {
         }
     }
 
-    public func feeRates() async throws -> [FeeRate] { fatalError("not implemented") }
+    public func feeRates() async throws -> [FeeRate] {
+        let (baseFee, priorityFees) = try await getBasePriorityFees(
+            blocks: Self.historyBlocks,
+            rewardPercentiles: EVMHistoryRewardPercentiles()
+        )
+
+        let feeRates: [FeeRate] = {
```

Comment:
no need to have a closure here? just do `var rates: [FeeRate] = []` and a for loop